ngraph/core/include/ngraph/op/util/nms_base.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace util
+        {
+            /// \brief Base class for operations NmsBase and MatrixNms
+            ///
+            class NGRAPH_API NmsBase : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                enum class SortResultType
+                {
+                    CLASSID, // sort selected boxes by class id (ascending) in each batch element
+                    SCORE,   // sort selected boxes by score (descending) in each batch element
+                    NONE     // do not guarantee the order in each batch element
+                };
+
+                NmsBase() = delete;
+
+                /// \brief Constructs a NmsBase operation
+                ///
+                /// \param output_type Specifies the output tensor type
+                /// \param nms_top_k Specifies maximum number of boxes to be selected per
+                /// class, -1 meaning to keep all boxes
+                /// \param keep_top_k Specifies maximum number of boxes to be selected per
+                /// batch element, -1 meaning to keep all boxes
+                NmsBase(ngraph::element::Type& output_type, int& nms_top_k, int& keep_top_k);
+
+                /// \brief Constructs a NmsBase operation
+                ///
+                /// \param boxes Node producing the box coordinates
+                /// \param scores Node producing the box scores
+                /// \param output_type Specifies the output tensor type
+                /// \param nms_top_k Specifies maximum number of boxes to be selected per
+                /// class, -1 meaning to keep all boxes
+                /// \param keep_top_k Specifies maximum number of boxes to be selected per
+                /// batch element, -1 meaning to keep all boxes
+                NmsBase(const Output<Node>& boxes,
+                        const Output<Node>& scores,
+                        ngraph::element::Type& output_type,
+                        int& nms_top_k,
+                        int& keep_top_k);
+
+                void validate_and_infer_types() override;
+
+                element::Type get_output_type() const { return m_output_type; }
```

Comment:
nitpick: we can return `const element::Type&`

Reply:
Done